cask: init at 0.8.4

FlexW opened the pull request:
###### Motivation for this change
This adds the version cask as packaged by Arch Linux AUR.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
cask: init at 0.8.4

bendlas commented:
Hello! Thanks for submitting this!

Right now, I get `error: attribute 'flexw' missing`, please add your information in [maintainers/maintainer-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix).

FlexW commented:
> Right now, I get `error: attribute 'flexw' missing`, please add your information in [maintainers/maintainer-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix).

I've added the information and rebased.

bendlas commented:
Already tested + used it yesterday. Thank you very much!

bendlas commented on pkgs/development/tools/cask/default.nix:
```diff
@@ -0,0 +1,43 @@
+    sha256 = "02f8bb20b33b23fb11e7d2a1d282519dfdb8b3090b9672448b8c2c2cacd3e478";
+  };
+
+  doCheck = true;
```
btw, a possible build phase would be to compile the `*.el` files to `*.elc`, which would improve startup speed.

bendlas commented on pkgs/development/tools/cask/default.nix:
```diff
@@ -0,0 +1,43 @@
+  doCheck = true;
+
+  dontBuild = true;
+
```
btw, a possible build phase would be to compile the `*.el` files to `*.elc`, which would improve startup speed.

FlexW replied:
Good idea. I tried it but byte compiling seems not be possible due errors caused by lexical variables.

adisbladis commented:
Why is this not using the sources from the generated melpa packages? 
That would mean that this package would get automatic updates.

cc @FlexW
